### ISSUE TITLE
Wrong IDE0004 Cast is redundant when applying attribute with params argument

### ISSUE THREAD

**Issue by JeppeSN** [dotnet/roslyn#20630]
**Version Used**: 
First seen in VS2015. Subsequently reproduced on:

> VS2017, version 15.2 (26430.14) Release

**Steps to Reproduce**:

Use this complete Console app:

```csharp
using System;
using System.Reflection;

class MarkAttribute : Attribute
{
  public readonly string[] Arr;

  public MarkAttribute(params string[] arr)
  {
    Arr = arr;
  }
}

[Mark((string)null)]   // wrong instance of: IDE0004 Cast is redundant.
static class Program
{
  static void Main()
  {
    var arr = typeof(Program).GetCustomAttribute<MarkAttribute>().Arr;
    Console.WriteLine(arr?.Length ?? int.MinValue);
  }
}
```

When compiled and run, correctly writes 1 to the console.

**Expected Behavior**:

No suggestion to remove cast syntax on null literal in attribute specification on class Program.

**Actual Behavior**:

Cast to string is in a pale text color, and a light bulb suggests, "Remove Unnecessary Cast". If the bad suggestion is followed, program behavior changes from writing 1 into writing -2147483648.

Obviously, the issue is with a params constructor that can apply in either normal form or expanded form.

While this seems _similar_ to other issues, like #18978, I do not know if it is a duplicate of any issue.

**Comment by sharwell** [comment]
:memo: I'm able to reproduce this issue even after #18978 is fixed.
